specs/agents/tracing-instrumentation-azure.md

```diff
@@ -0,0 +1,420 @@
+| [Azure China](https://docs.microsoft.com/en-us/azure/china/resources-developer-guide) |`<Storage Account Name>.table.core.chinacloudapi.cn` |
+| [Azure Germany](https://docs.microsoft.com/en-us/azure/germany/germany-developer-guide) | `<Storage Account Name>.table.core.cloudapi.de` |
+
+where `<Storage Account Name>` is the name of the storage account. New Azure service endpoints may be introduced by Azure later.
+
+Rules derived from the [Table service REST API reference](https://docs.microsoft.com/en-us/rest/api/storageservices/table-service-rest-api)
+
+| URL | HTTP verb | HTTP headers | HTTP query string | Resulting Operation Name |
+| --- | --------- | ---------- | ------------------- | ------------------------ |
+| | PUT    |            |   `comp=properties`                  | SetProperties    |
+| | GET    |            |   `comp=properties`                  | GetProperties    |
+| | GET    |            |   `comp=stats`                       | Stats            |
+| ends with /Tables | GET    |            |                  | Query           |
+| ends with /Tables | POST    |            |                 | Create           |
+| ends with /Tables('`<table>`') | DELETE    |            |  | Delete           |
+| ends with /`<table>` | OPTIONS    |            |           | Preflight        |
+| | HEAD      |            | `comp=acl`          | GetAcl                   |
+| | GET      |            | `comp=acl`          | GetAcl                   |
+| | PUT      |            | `comp=acl`          | SetAcl                   |
+| ends with /`<table>`() or /`<table>`(PartitionKey='`<partition-key>`',RowKey='`<row-key>`')`  | GET    |            |                  | Query            |
+| ends with /`<table>` | POST      |            |           | Insert                   |
+| ends with /`<table>`(PartitionKey='`<partition-key>`',RowKey='`<row-key>`')` | PUT      |            |           | Update                   |
+| ends with /`<table>`(PartitionKey='`<partition-key>`',RowKey='`<row-key>`')` | MERGE      |            |           | Merge                   |
+| ends with /`<table>`(PartitionKey='`<partition-key>`',RowKey='`<row-key>`')` | DELETE      |            |           | Delete                   |
+
+### File share storage
+
+Azure file service (known also as AFS or file share storage) is a managed file share service
+that uses Server Message Block (SMB) protocol or Network File System (NFS) protocol
+to provide general purpose file shares. File shares can be mounted concurrently by
+machines in the cloud or on-premises.
+
+The `<ResourceName>` is determined from the path of the URL.
+
+`<Storage Account Name>` is the leftmost subdomain of the domain of an Azure storage account endpoint.
+
+| APM field | Required? | Format | Notes | Example |
+| --------- | --------- | ------ | ----- | ------- |
+| `span.name` | yes | `AzureFile <OperationName> <ResourceName>` | Pascal case Operation name | `AzureFile Create directoryname` |

```

Comment:
@felixbarny thoughts on excluding the `<ResourceName>` from the span name? If it were to be removed, is there another appropriate field to capture it?